### PR TITLE
EKF: Convert EKF-GSF yaw estimator from matlab to sympy derived equations

### PR DESCRIPTION
**Pull request by priseborough**
Tested in SITL with horizontal position changes using standard gazebo MC model. Replay of the log was then used to compare the behaviour of the EKF-GSF yaw estimator before and after. There was no visible difference in outputs. note that this includes two patches from https://github.com/PX4/ecl/pull/870 that add the custom ecl::powf function

EKF-GSF estimates and weights before:

Screen Shot 2020-07-23 at 9 26 44 pm

After changes:

Screen Shot 2020-07-23 at 9 30 55 pm

### CODE REVIEW

**Comment by kamilritz** [comment]
@priseborough Found a nice way to optimize the auto generated measurement update at once. This avoids some computation duplication and checking for the same division multiple times. Now we directly auto generate the computation of the inverse innovation covariance, therefore we get rid of the `updateInnovCovMatInv` function too.

I took the liberty of adding that to this PR, I hope that is okay.

**Comment by priseborough** [comment]
@kamilritz Thanks for that. I've run the replay before and after your imrpvements and the results are visually identical, eg:

Before:

Screen Shot 2020-07-30 at 8 04 20 am


After:

Screen Shot 2020-07-30 at 8 02 50 am

File: EKF/python/gsf_ekf_yaw_estimator/code_gen.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (priseborough) [reply EKF/python/gsf_ekf_yaw_estimator/code_gen.py:1]: Not exactly the same at the moment, but there are no required output formatting differences, so as soon as https://github.com/PX4/ecl/pull/870 is merged, we can remove it.

**Comment by priseborough** [comment]
@kamilritz I've been unable to get python to import code_gen from the ekf_derivation directory.r

File: EKF/EKFGSF_yaw.cpp [hunk 2]
```diff
@@ -365,80 +358,80 @@ bool EKFGSF_yaw::updateEKF(const uint8_t model_index)
...
+	const float SP3 = P00 + velObsVar;
+	const float SP4 = SP0*SP3;
+	const float SP5 = SP2 + SP4;
+	if (fabsf(SP5) < 1e-3f) {
+		return false;
+	}
+	const float SP6 = 1.0F/SP5;
+	const float SP7 = SP6*velObsVar;
+	// const float SP8 = (-P00*SP0 + SP1)/(SP1 - SP4);
+	const float SP8_denom = SP1 - SP4;
```
[the remark this answers is not in the document]
Comment (priseborough) [reply EKF/EKFGSF_yaw.cpp]: fixed

**Pull request merged**